pkg/util/rollout_utils.go

```diff
@@ -155,6 +151,12 @@ func AddWatcherDynamically(c controller.Controller, h handler.EventHandler, gvk
 	return true, c.Watch(&source.Kind{Type: object}, h)
 }
 
+func HashReleasePlanBatches(releasePlan *rolloutv1alpha1.ReleasePlan) string {
```

Comment:
can we use the generation field of the BatchRelease, hash the spec can be problematic if we add fields to the ReleasePlan